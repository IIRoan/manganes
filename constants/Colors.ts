// constants/Colors.ts

export const Colors = {
  light: {
    primary: '#2E8B57',
    background: '#F5F5F5',
    card: '#FFFFFF',
    text: '#333333',
    border: '#E0E0E0',
    notification: '#FF6B6B',
    tint: '#2E8B57',
    tabIconDefault: '#BDBDBD',
    tabIconSelected: '#2E8B57',
    secondaryText: '#757575',
    error: '#FF6B6B',

  },
  dark: {
    primary: '#4CAF50',
    background: '#121212',
    card: '#1f1f1d',
    text: '#E0E0E0',
    border: '#333333',
    notification: '#FF6B6B',
    tint: '#4CAF50',
    tabIconDefault: '#757575',
    tabIconSelected: '#4CAF50',
    secondaryText: '#BDBDBD',
    error: '#FF6B6B',
  },
};

export type ColorScheme = keyof typeof Colors;
export type ThemeColors = typeof Colors.light & typeof Colors.dark;
