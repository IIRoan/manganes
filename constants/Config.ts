export const MANGA_API_URL = 'https://mangafire.to';
