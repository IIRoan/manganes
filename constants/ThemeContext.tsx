import React, { createContext, useState, useContext, useEffect } from 'react';
import { useColorScheme } from 'react-native';
import { ColorScheme } from '@/constants/Colors';

type Theme = 'light' | 'dark' | 'system';

interface ThemeContextType {
  theme: Theme;
  systemTheme: ColorScheme;
  setTheme: (theme: Theme) => void;
  toggleTheme: () => void;
  actualTheme: 'light' | 'dark';
}

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [theme, setTheme] = useState<Theme>('system');
  const systemColorScheme = useColorScheme() as ColorScheme;

  const toggleTheme = () => {
    setTheme(prevTheme => {
      switch (prevTheme) {
        case 'light': return 'dark';
        case 'dark': return 'system';
        case 'system': return 'light';
      }
    });
  };

  const actualTheme = theme === 'system' ? systemColorScheme : theme;

  return (
    <ThemeContext.Provider value={{ 
      theme, 
      systemTheme: systemColorScheme, 
      setTheme,
      toggleTheme,
      actualTheme
    }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const context = useContext(ThemeContext);
  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};