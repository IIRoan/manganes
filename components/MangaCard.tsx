import React from 'react';
import { View, Text, Image, TouchableOpacity, StyleSheet } from 'react-native';
import { Colors, ColorScheme } from '@/constants/Colors';
import { useTheme } from '@/constants/ThemeContext';

interface MangaCardProps {
  title: string;
  imageUrl: string;
  onPress: () => void;
}

const MangaCard: React.FC<MangaCardProps> = ({ title, imageUrl, onPress }) => {
  const { theme, systemTheme } = useTheme();
  const colorScheme = theme === 'system' ? systemTheme : theme as ColorScheme;
  const colors = Colors[colorScheme];

  return (
    <TouchableOpacity 
      onPress={onPress} 
      style={[
        styles.card, 
        { 
          backgroundColor: colors.card,
          borderColor: colors.border,
        }
      ]}
    >
      <Image source={{ uri: imageUrl }} style={styles.image} />
      <View style={styles.titleContainer}>
        <Text 
          style={[
            styles.title, 
            { color: colors.text }
          ]} 
          numberOfLines={2}
        >
          {title}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  card: {
    width: '48%',
    marginBottom: 10,
    borderRadius: 8,
    overflow: 'hidden',
    borderWidth: 1,
  },
  image: {
    width: '100%',
    height: 200,
    resizeMode: 'cover',
  },
  titleContainer: {
    padding: 8,
  },
  title: {
    fontSize: 14,
    fontWeight: 'bold',
  },
});

export default MangaCard;