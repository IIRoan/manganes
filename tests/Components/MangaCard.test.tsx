import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import MangaCard from '../../components/MangaCard';
import { useTheme } from '@/constants/ThemeContext';

jest.mock('@/constants/ThemeContext', () => ({
    useTheme: jest.fn(),
}));

describe('MangaCard Component', () => {
    const onPressMock = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();
        (useTheme as jest.Mock).mockReturnValue({
            theme: 'light',
            systemTheme: 'light',
        });
    });

    it('renders correctly with all props', () => {
        const { getByText, getByLabelText } = render(
          <MangaCard
            title="Manga Title"
            imageUrl="http://example.com/image.jpg"
            onPress={onPressMock}
            lastReadChapter="Chapter 5"
          />
        );
      
        expect(getByText('Manga Title')).toBeTruthy();
        expect(getByText('Last read: Chapter 5')).toBeTruthy();
        expect(getByLabelText('Manga Image')).toBeTruthy();
      });
      


    it('calls onPress when pressed', () => {
        const { getByTestId } = render(
            <MangaCard
                title="Manga Title"
                imageUrl="http://example.com/image.jpg"
                onPress={onPressMock}
                lastReadChapter="Chapter 5"
            />
        );

        fireEvent.press(getByTestId('manga-card'));
        expect(onPressMock).toHaveBeenCalled();
    });


    it('renders correctly without lastReadChapter', () => {
        const { queryByText } = render(
            <MangaCard
                title="Manga Title"
                imageUrl="http://example.com/image.jpg"
                onPress={onPressMock}
                lastReadChapter={null}
            />
        );

        expect(queryByText(/Last read:/)).toBeNull();
    });

    it('matches snapshot', () => {
        const tree = render(
            <MangaCard
                title="Snapshot Manga"
                imageUrl="http://example.com/image.jpg"
                onPress={onPressMock}
                lastReadChapter="Chapter 10"
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
