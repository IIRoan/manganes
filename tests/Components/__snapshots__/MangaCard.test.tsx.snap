// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MangaCard Component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 12,
      "opacity": 1,
      "overflow": "hidden",
    }
  }
  testID="manga-card"
>
  <Image
    accessibilityLabel="Manga Image"
    source={
      {
        "uri": "http://example.com/image.jpg",
      }
    }
    style={
      {
        "aspectRatio": 0.75,
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "padding": 8,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        {
          "color": "#333333",
          "fontSize": 14,
          "fontWeight": "600",
          "marginBottom": 4,
        }
      }
    >
      Snapshot Manga
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        {
          "color": "#BDBDBD",
          "fontSize": 12,
        }
      }
    >
      Last read: 
      Chapter 10
    </Text>
  </View>
</View>
`;
